### PR TITLE
wgengine/netstack: add test for #12448

### PR DESCRIPTION
**Pull request by andrew-d**
This refactors the logic for determining whether a packet should be sent to the host or not into a function, and then adds tests for it.

Updates #11304
Updates #12448

Change-Id: Ief9afa98eaffae00e21ceb7db073c61b170355e5

### CODE REVIEW

File: wgengine/netstack/netstack.go [hunk 1]
```diff
@@ -866,6 +826,53 @@ func (ns *Impl) inject() {
...
+					return true
+				}
+				if debugNetstack() {
+					ns.logf("netstack: sending 4via6 packet to host: src=%v dst=%v", srcIP, dstIP)
+				}
+			}
+		}
+	default:
+		// unknown; don't forward to host
+	}
```
Comment (irbekrm) [inline wgengine/netstack/netstack.go:871]: I guess we don't really need this as the `return false` at the end cover this case

File: wgengine/netstack/netstack.go [hunk 2]
```diff
@@ -866,6 +826,53 @@ func (ns *Impl) inject() {
 	}
 }
 
+func (ns *Impl) shouldSendToHost(pkt *stack.PacketBuffer) bool {
+	// Determine if the packet is from a service IP, in which case it
```
Comment (irbekrm) [inline wgengine/netstack/netstack.go:830]: Nit: for me reading this first time it was not immediately clear what 'service' is in this context

**Review by irbekrm — APPROVED** [review]
Nice, thank you for adding the tests! I'm paranoid, so I also built and tested this change.
Added two optional nits

File: wgengine/netstack/netstack.go (hunk 1 above)
Comment (andrew-d) [reply wgengine/netstack/netstack.go:871]: Oh, good catch; I was meaning to add a log message here.

File: wgengine/netstack/netstack.go (hunk 2 above)
Comment (andrew-d) [reply wgengine/netstack/netstack.go]: Thanks; good point. Clarified with a comment.

**Pull request merged**